src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -64,12 +66,19 @@ public async Task Invoke(HttpContext context)
                 return;
             }
 
-            // Policy evaluator has transient lifetime so it fetched from request services instead of injecting in constructor
+            // Policy evaluator has transient lifetime so it's fetched from request services instead of injecting in constructor
             var policyEvaluator = context.RequestServices.GetRequiredService<IPolicyEvaluator>();
 
             var authenticateResult = await policyEvaluator.AuthenticateAsync(policy, context);
 
-            // Allow Anonymous skips all authorization
+            if (authenticateResult?.Succeeded ?? false)
+            {
+                var authFeatures = new AuthenticationFeatures(authenticateResult);
+                context.Features.Set<IHttpAuthenticationFeature>(authFeatures);
+                context.Features.Set<IAuthenticateResultFeature>(authFeatures);
```

Comment:
Can this check for the existing IAuthenticateResultFeature and re-use it? Otherwise you'll end up allocating this feature twice on many requests.

Reply:
Lol, glad you see my point of view now
>Main issue is we'll need to write some ugly code in Authorization middleware to check if the AuthenticationFeatures class has been set on the features by the Authentication middleware.